libs/utils/energy.py

```diff
@@ -118,6 +121,58 @@ def reset(self):
     def report(self, out_dir):
         raise NotImplementedError('Missing implementation')
 
+class Monsoon(EnergyMeter):
+    def print_log(self, errmsg, exception=0):
+	print errmsg
+	self._log.error(errmsg)
+	if exception == 1:
+		raise(Exception(errmsg))
+
+    def __init__(self, target, conf=None):
+	super(Monsoon, self).__init__(target, '')
+
+	# Read config
+	# { 'config': { 'device_entry' : '/dev/ttyACM1', 'voltage' : 4.3 } }
+	try:
+		device_entry = conf['config']['device_entry']
+		self.voltage = conf['config']['voltage']
+	except:
+		device_entry = None
+		self.voltage = 4.3
+		self._log.info("Invalid conf, using defaults. Example format is \
+			       { 'config': { 'device_entry' : '/dev/ttyACM1', 'voltage' : 4.3 }")
+	self._monsoon = devlib.MonsoonInstrument(self._target, device_entry)
+
+    def reset_only(self):
+	m = self._monsoon
+        self._log.info('Initiailzing Monsoon')
+	# Waiting for monsoon to get ready
+	m.get_status()
+	# Provide power to the target
+	m.set_voltage(0)
+	m.set_current(8)
+	m.set_start_current(8)
+	m.set_voltage(self.voltage)
+	self._log.info("Monsoon ready for action")
+
+    # Start a monsoon run asynchronously
+    def reset(self):
+	m = self._monsoon
+	self.reset_only()
+	m.start()
+
+    # Kill an already running test and report in panda dataframes
+    def report(self):
+	return self._monsoon.report()
+
+    # This is synchronous (block until duration)
+    def get_samples(self, duration):
```

Comment:
LISA's EnergyMeters API expect to provide a really simple interface where the user needs only two methods:
- reset: which start a new energy measurement session
- report: which reports the energy measured since last reset and _all the additional data produced by the specific EM_

This means that you should better put whatever this code produce at the end of the report call.

You should have a look at the ACME's report function at the end of this module.

NOTE: perhaps this can require to have a different implementation of the devlib's Monsoon module's get_samples_sync(), which BTW I'm not sure it is an "expected" interface for the devlib's Instrument module.
We should implement if using either the "instantaneous" or the "continuous" APIs:
https://github.com/ARM-software/devlib/blob/master/devlib/instrument/__init__.py#L218